File: dcp-solutions/2024-11-07/solution.rb
```ruby
def search(nums, target):
  if nums.empty?
    return nil
  end
  
  left, right = 0, nums.size - 1
  while left <= right do
    mid = (left + right) / 2
    
    # If the target is found at index mid
    if nums[mid] == target
      return mid
    # The target must be in the half of the array on the left side
    elif nums[left] <= nums[mid]
      # If the target is in the range [nums[left], nums[mid]]
      if nums[left] <= target && target < nums[mid]
        return mid
      end
      # Move to the left half
      right = mid - 1
    else
      # The target must be in the half of the array on the right side
      # If the target is in the range [nums[mid], nums[right]]
      if nums[mid] < target && target <= nums[right]
        return mid
      end
      # Move to the right half
      left = mid + 1
    end
  end
  
  # If the target is not found
  return nil
end
```

File: dcp-solutions/2024-11-07/solution.php
```php
class Solution {
    function search($nums, $target) {
        if (empty($nums)) return null;

        $left = 0;
        $right = count($nums) - 1;

        while ($left <= $right) {
            $mid = floor(($left + $right) / 2);

            if ($nums[$mid] == $target) return $mid;

            // Determine the left and right bound of the search space
            if ($nums[$left] <= $nums[$mid]) {
                if ($nums[$left] <= $target && $target < $nums[$mid])
                    $right = $mid - 1;
                else
                    $left = $mid + 1;
            } else {
                if ($nums[$mid] < $target && $target <= $nums[$right])
                    $left = $mid + 1;
                else
                    $right = $mid - 1;
            }
        }

        return null;
    }
}
```